File: hospital-backend/src/main/java/com/hospital/dao/RoomDAO.java
package com.hospital.dao;

import com.hospital.model.Room;
import java.util.List;

public interface RoomDAO {
    List<Room> getAll();
    Room getById(int id);
    int  insert(Room room);
    int  update(Room room);
    int  delete(int id);
}

File: hospital/model/dao/RoomDAO.js
/**
 * DAO Interface: RoomDAO
 */
export default class RoomDAO {
  async getAll()             { throw new Error('RoomDAO.getAll() not implemented'); }
  async getById(id)          { throw new Error('RoomDAO.getById() not implemented'); }
  async getAvailable()       { throw new Error('RoomDAO.getAvailable() not implemented'); }
  async getByType(type)      { throw new Error('RoomDAO.getByType() not implemented'); }
  async create(room)         { throw new Error('RoomDAO.create() not implemented'); }
  async update(id, data)     { throw new Error('RoomDAO.update() not implemented'); }
  async delete(id)           { throw new Error('RoomDAO.delete() not implemented'); }
  async setAvailability(id, isAvailable) { throw new Error('RoomDAO.setAvailability() not implemented'); }
}
